### PR TITLE
enable and fix e2e tests for new SDK

### PR DESCRIPTION
**Pull request by nylira**
Let's work together to fix the E2E tests for the new SDK!

> Jordan:
> TLDR: i’m pretty sure the e2e tests are failing because “sign in” is hidden from the welcome modal - because our `createAccount` function in our e2e tests doesn’t actually create accounts - it recovers them - and recover may have changed a bit in `gaiacli` - or at least is not working as expected ATM

- [x] passing e2e tests

### CODE REVIEW

File: app/src/main/index.js [hunk 1]
```diff
@@ -232,6 +239,8 @@ async function startLCD(home, nodeIP) {
       reject()
       afterBooted(() => {
         if (mainWindow) {
+          // TODO unify/refactor logError and webContents.send
+          logError(`The ${SERVER_BINARY} rest-server (LCD) exited unplanned`)
```
Comment (jolesbi) [inline app/src/main/index.js]: i know this is not a change from this PR - but why do we call this `SERVER_BINARY`?

File: .circleci/config.yml [hunk 2]
```diff
@@ -35,9 +35,9 @@ jobs:
       - run:
           name: Add Gaia
           command: |
-            sudo wget https://www.dropbox.com/s/lyzwloe4swr5is5/gaia-0.13.0-linux?dl=0 -O ~/repo/gaia
+            sudo wget https://www.dropbox.com/s/c82lrtttdiqtmki/gaiacli-0.16.0-linux?dl=0 -O ~/repo/gaia
```
[the remark this answers is not in the document]
Comment (nylira) [reply .circleci/config.yml]: We need to wait on Greg for an official solution but he’s ok vacation. I think this is an acceptable compromise for now.

File: app/src/main/index.js [hunk 3]
```diff
@@ -589,18 +598,18 @@ async function main() {
   log(`winURL: ${winURL}`)
 
   // XXX: currently ignores commit hash
-  let basecliVersion = (await getBasecoindVersion()).split(" ")[0]
+  let gaiacliVersion = (await getBasecoindVersion()).split("-")[0]
```
Comment (jolesbi) [inline app/src/main/index.js]: are we actually using `basecoind` anymore? i thought we are using `gaiad`...

File: test/e2e/launch.js [hunk 4]
```diff
@@ -11,9 +11,10 @@ const shell = require(`shelljs`)
 
 const networkPath = join(__dirname, "localtestnet")
 
-let app, home, nodeHome, started
-let binary = process.env.BINARY_PATH || "$GOPATH/bin/gaiacli"
-let nodeBinary = process.env.NODE_BINARY_PATH
+let app, cliHome, nodeHome, started
+let binary = process.env.BINARY_PATH || process.env.GOPATH + "/bin/gaiacli"
```
Comment (jolesbi) [inline test/e2e/launch.js]: for my own learning, why is it better to use `process.env.GOPATH` over `$GOPATH`?

File: app/src/main/index.js (hunk 1 above)
Comment (nylira) [reply app/src/main/index.js]: I guess it could be called LCD_BINARY. The SERVER part must refer to the REST server.

File: test/e2e/launch.js [hunk 5]
```diff
@@ -43,12 +44,13 @@ function launch(t) {
       })
 
     started = new Promise(async (resolve, reject) => {
-      console.log("using binary", binary)
+      console.log("using cli binary", binary)
```
Comment (jolesbi) [inline test/e2e/launch.js]: is this always going to refer to `gaiacli`?

File: test/e2e/launch.js [hunk 6]
```diff
@@ -43,12 +44,13 @@ function launch(t) {
...
+      console.log("using node binary", nodeBinary)
 
       // TODO cleanup
-      home = newTempDir()
+      cliHome = newTempDir()
```
Comment (jolesbi) [inline test/e2e/launch.js]: what is `cliHome`?

File: app/src/main/index.js (hunk 1 above)
Comment (jolesbi) [reply app/src/main/index.js]: i think a name change could reduce confusion. too many combinations of `server, cli, binary` to keep track of.

File: test/e2e/launch.js (hunk 4 above)
Comment (faboweb) [reply test/e2e/launch.js]: `$GOPATH` is not necessarily resolved

File: test/e2e/launch.js (hunk 5 above)
Comment (faboweb) [reply test/e2e/launch.js]: Yes, but the path to `gaiacli`

File: test/e2e/launch.js (hunk 6 above)
Comment (faboweb) [reply test/e2e/launch.js]: `gaiacli --home XXX`
It stores the config files for the cli and the keys

**Comment by nylira** [comment]
Well done @faboweb!

**Pull request merged**